### ISSUE TITLE
choose a better set of default repositories on sourcegraph.com

### ISSUE THREAD

**Issue by ijt** [sourcegraph/sourcegraph#4992]
BigQuery can be used to find popular github repositories using the `githubarchive` tables, like this:
```
$ cat /tmp/q
SELECT
repo.name, COUNT(*) AS cnt
FROM
    [githubarchive:year.2018]
WHERE type = 'WatchEvent'
GROUP BY repo.name,
ORDER BY cnt DESC
LIMIT 1000

$ bq query --max_rows=1000 --format=csv "$(cat /tmp/q)" > /tmp/tr.csv                                                                                                       
```
This contains several famous repositories within the first 600:
```
[ ~ ] egrep '(torvalds/linux|vim/vim|/react|/tensorflow|/angular|/sourcegraph|/kubernetes),'  <(head -n 600 /tmp/tr.csv)
facebook/react,35686
tensorflow/tensorflow,32774
kubernetes/kubernetes,14475
torvalds/linux,14347
angular/angular,12549
sourcegraph/sourcegraph,4281
vim/vim,3943
```

~We could use the results of this query by introducing a new table called something like `repos_to_index`, uploading the bq results to it, and joining against it when listing the repos to index on sourcegraph.com.~

**Comment by keegancsmith** [comment]
Sounds good to me.

**Comment by ijt** [comment]
This supports https://github.com/sourcegraph/sourcegraph/issues/4958.

**Comment by ijt** [comment]
I thought of a more general solution that doesn't involve a special table or any other special treatment for sourcegraph.com: add an indexed `score` column to the `repo` table so we have a way to keep track of how important the repos are. The watch count would be a fine score to start with. We could add a `repolimit` field to decide how many repos to include, defaulting to `600` or so (configurable). 

@keegancsmith, @tsenart WDYT?

**Comment by tsenart** [comment]
@ijt: A `score` column seems to belong to the search domain, but the repo table itself isn't. Additionally, it's best that we reduce additional write load going to this table, so I'd prefer that such a watch count / score be kept in a separate table.

**Comment by keegancsmith** [comment]
The benefit of populating another table is we have much more control over what exactly we index. EG we could be flexible and include all repos with a certain watch count + some hand picked repositories quite easily. Additionally the decision of what to index is decoupled from the request asking for the list of repositories to index. Another trade-off in favour of a separate table is from a performance perspective sourcegraph.com's repo table is very large, while the set of repos we actually want to index is quite small (however postgres does have partial indexes).

However, the use of a watch or star count is quite simple. If we do decide to go that route there is quite a nice way to do it. Include those values in the metadata we store. Then you can get postgres to index just those values in the metadata. 

The above would be sourcegraph.com and github specific. So a separate table seems like the more general and flexible design.

**Comment by ijt** [comment]
@tsenart, sure, separating the concerns makes sense. The argument about write load is surprising since I don't think there would typically be a lot of writes on the `repo` table, especially not ones whose performance would be affected by having a new column like this.

**Comment by ijt** [comment]
Here is a document I wrote describing a solution involving a `default_repos` table:
https://www.notion.so/Default-repositories-for-search-c2003d1d4c18473c992fc69116f9616d

**Comment by ijt** [comment]
This issue was fixed by #5062.
